pkg/providers/v2/cloud.go

```diff
@@ -30,12 +32,26 @@ import (
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ec2"
 
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	cloudprovider "k8s.io/cloud-provider"
+	"k8s.io/cloud-provider-aws/pkg/apis/config/v1alpha1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
 )
 
 func init() {
 	cloudprovider.RegisterCloudProvider(ProviderName, func(config io.Reader) (cloudprovider.Interface, error) {
-		return newCloud()
+		cfg, err := readAWSCloudConfig(config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read AWS cloud provider config file: %v", err)
+		}
+
+		errs := validateAWSCloudConfig(cfg)
```

Comment:
@andrewsykim updated! PTAL :) 
- added validation function & unit tests
- update the Instances implementation to use the clusterName tags for API calls